app/email_branding/rest.py

```diff
@@ -73,3 +76,30 @@ def update_email_branding(email_branding_id):
     dao_update_email_branding(fetched_email_branding, **data)
 
     return jsonify(data=fetched_email_branding.serialize()), 200
+
+
+@email_branding_blueprint.route("/get-name-for-alt-text/", methods=["POST"])
+def get_email_branding_name_for_alt_text():
+    data = request.get_json()
+
+    validate(data, post_get_email_branding_name_for_alt_text_schema)
+
+    alt_text = data["alt_text"]
+
+    existing_branding = dao_get_email_branding_by_name_case_insensitive(alt_text)
+    if not existing_branding:
+        chosen_name = alt_text
+    else:
+        existing_alternate_branding_options = [
+            x.name for x in dao_get_existing_alternate_email_branding_for_name(alt_text)
+        ]
```

Comment:
optional: this would be nicer as a set than a list

Reply:
very uncharacteristic of me to miss an opportunity to use a set. squashed in to the commit